src/model/document.ts

```diff
@@ -238,10 +191,45 @@ export class Document implements Feature, Queryable {
     this._localFeatures.add(feature);
   }
 
-  getByKind<K extends keyof FeatureKinds>(kind: K, options?: QueryOptions):
+  /**
+   * Get features on the document.
+   *
+   * Be default it includes only features on the document, but you can specify
+   * whether to include features that are reachable by imports, features from
+   * outside the current package, etc. See the documentation for Query for more
+   * details.
+   *
+   * You can also narrow by feature kind and identifier.
+   */
+  getFeatures<K extends keyof FeatureKinds>(query: QueryWithKind<K>):
       Set<FeatureKinds[K]>;
-  getByKind(kind: string, options?: QueryOptions): Set<Feature>;
-  getByKind(kind: string, options?: QueryOptions): Set<Feature> {
+  getFeatures(query: Query): Set<Feature>;
+  getFeatures(query: Query): Set<Feature> {
+    const queryOptions = {
```

Comment:
it seems like you're splitting the query up into separate objects, then mostly passing them together. Why not make `_getById`, `_getByKind` and `listFeatures` take a `Query` object?